pandas/tests/io/test_stata.py

```diff
@@ -1144,7 +1143,7 @@ def _convert_categorical(from_frame: DataFrame) -> DataFrame:
             if is_categorical_dtype(ser.dtype):
                 cat = ser._values.remove_unused_categories()
                 if cat.categories.dtype == object:
-                    categories = ensure_index(cat.categories._values)
+                    categories = pd.Index._with_infer(cat.categories._values)
```

Comment:
Is this supposed to have _with_infer still?

Reply:
yes